[1.21.1] Fix Processing Output not applying data components

RaymondBlaze opened the pull request:
Also optimized `rollOutput` for case when chance is 100%, and added count / chance validation in codec.

See https://github.com/Creators-of-Create/Create/pull/7945#issuecomment-2748418994.

RaymondBlaze committed:
Fix ProcessingOutput not applying components

RaymondBlaze committed:
Optimize rollOutput for 100% chance case

RaymondBlaze committed:
Add validation for ProcessingOutput's count and chance

RaymondBlaze committed:
Remove validation for chance bigger than 1 (Sequenced Assembly use them as weight)

lcy0x1 reviewed (changes requested):
`getStack()` still doesn't have the patch applied

RaymondBlaze committed:
Fix getStack and use getStack for rollOutput
Add overload for datagenOutput with components

RaymondBlaze commented:
> `getStack()` still doesn't have the patch applied

Fixed, also removed the behaviour of `getStack()` checking for `datagenOutput`, since it's not used in datagen and there's no point to do so.

IThundxr commented:
Thanks!
